### PR TITLE
Week2 googlemaps and fontend

### PR DESCRIPTION
**Pull request by mchristia**
Major changes in frontend
- add google maps
- add pin points for all surf spots in google maps
- show information for spots in google maps
- receive surf spot data in details page
- add table with data in details page

### CODE REVIEW

File: backend/src/main/java/de/neuefische/backend/service/SGApiService.java [hunk 1]
```diff
@@ -51,7 +51,7 @@ public SurfSpot getSGData(String longitude, String latitude){
     public String generateRequestString(String longitude, String latitude){
         //Instant.now().plus(3, ChronoUnit.HOURS);
```
Comment (jamarob) [inline backend/src/main/java/de/neuefische/backend/service/SGApiService.java]:
```suggestion

```

File: backend/src/main/java/de/neuefische/backend/service/SGApiService.java [hunk 2]
```diff
@@ -51,7 +51,7 @@ public SurfSpot getSGData(String longitude, String latitude){
...
         startTimeStampForSGRequest = generateStartTime(); //Auf genaue Stunde abrunden
```
Comment (jamarob) [inline backend/src/main/java/de/neuefische/backend/service/SGApiService.java]: please remove comments. Better keep `startTimeStampForSGRequest` and `endTimeStampForSGRequest` local to the method.

File: backend/src/main/java/de/neuefische/backend/service/SGApiService.java [hunk 3]
```diff
@@ -62,8 +62,12 @@ public String generateRequestString(String longitude, String latitude){
                 "&start="+ startTimeStampForSGRequest +
                 "&end=" + endTimeStampForSGRequest +"&source=sg";
     }
-
+// Start time should always be 00:00 o'clock the day of the request!
```
Comment (jamarob) [inline backend/src/main/java/de/neuefische/backend/service/SGApiService.java]: Please delete comments

File: frontend/src/App.js [hunk 4]
```diff
@@ -3,25 +3,26 @@ import HomePage from "./pages/HomePage";
 import SearchPage from "./pages/SearchPage";
 import SearchMapPage from "./pages/SearchMapPage";
 import SpotDetailsPage from "./pages/SpotDetailsPage";
+import useSurfSpot from "./hooks/useSrufSpots";
```
Comment (jamarob) [inline frontend/src/App.js]: Typo in file name

File: frontend/src/component/DeatailsTable.js [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+
+const StyledTableRow = withStyles((theme) => ({
+    root: {
+        '&:nth-of-type(odd)': {
+            backgroundColor: theme.palette.action.hover,
+        },
+    },
+}))(TableRow);
+
+function createData(date, height, period, direction, windspeed, winddirection) {
```
Comment (jamarob) [inline frontend/src/component/DeatailsTable.js:33]: Is this function really necessary? Inlining `{ date, height, period, direction, windspeed, winddirection}` will do just fine

File: frontend/src/component/DeatailsTable.js [hunk 6]
```diff
@@ -0,0 +1,90 @@
...
+        overflowY: 'auto'
+    },
+});
+
+
+export default function DetailTable({surfSpot}){
+
+    const classes = useStyles();
+    const rows = []
+    console.log(new Date(surfSpot?.surfData[0].time).toDateString())
```
Comment (jamarob) [inline frontend/src/component/DeatailsTable.js:49]:
```suggestion

```

File: frontend/src/component/DeatailsTable.js [hunk 7]
```diff
@@ -0,0 +1,90 @@
...
+
+    surfSpot?.surfData.forEach(element => {
+
+        rows.push(createData(new Date(element.time).toDateString(),
+            element.waveHeight.sg,
+            element.wavePeriod.sg,
+            element.waveDirection.sg,
+            element.windSpeed.sg,
+            element.windDirection.sg))
+    })
```
Comment (jamarob) [inline frontend/src/component/DeatailsTable.js:59]: Use `.map`

File: frontend/src/component/DeatailsTable.js [hunk 8]
```diff
@@ -0,0 +1,90 @@
...
+    console.log(rows)
```
Comment (jamarob) [inline frontend/src/component/DeatailsTable.js:60]:
```suggestion

```

File: frontend/src/component/DetailsHeader.js [hunk 9]
```diff
@@ -0,0 +1,57 @@
...
+                {surfSpot?.spotLocationDetails.region}
+            </div>
+            <div>
+                {new Date(now).toUTCString()}
+            </div>
+            <div>
+                <label>
+                    Ait Temperature:
+                </label>
+                {rightTimeToShowCurrentTemp()?.airTemperature.sg}
```
Comment (jamarob) [inline frontend/src/component/DetailsHeader.js:18]: I don't get what this name means and it suggests a boolean, but it is a function that returns an object. If the function takes no parameters, why don't you directly pass in the value or undefined?

File: frontend/src/component/FavouritesList.js [hunk 10]
```diff
@@ -1,9 +1,9 @@
-import SpotItem from "./SpotItem";
+import ListItem from "./ListItem";
 
 export default function FavouritesList(){
     return(
         <div>
-            <SpotItem/>
+            <ListItem/>
```
Comment (jamarob) [inline frontend/src/component/FavouritesList.js:6]: `ListItem` is not a list item and not inside a list.. 🤔 maybe the naming could be better

File: frontend/src/component/ListItem.js [hunk 11]
```diff
@@ -0,0 +1,94 @@
...
+import Card from "@material-ui/core/Card"
+import red from '@material-ui/core/colors/red';
+import {Link} from "react-router-dom";
+
+// const useStyles = makeStyles(() => ({
+//     root:{
+//         minWidth: 275,
+//         outlineColor: "bisque",
+//         outlineWidth: 2,
+//     }}))
```
Comment (jamarob) [inline frontend/src/component/ListItem.js:12]:
```suggestion

```

File: frontend/src/component/ListItem.js [hunk 12]
```diff
@@ -0,0 +1,94 @@
...
+            flexWrap: "nowrap"
+        }
+    });
+
+
+
+export default function ListItem({spot}){
+    const classes = useStyles();
+
+    console.log(spot)
```
Comment (jamarob) [inline frontend/src/component/ListItem.js:50]:
```suggestion

```

File: frontend/src/component/ListItem.js [hunk 13]
```diff
@@ -0,0 +1,94 @@
...
+//       color: black;
+//       display: flex;
+//       flex-direction: column;
+//       background: lightblue;
+//       border: darkgrey 2px solid;
+//       border-radius: 6px;
+//       padding: 12px;
+//       margin: 12px;
+//       text-align: center;
+// `
```
Comment (jamarob) [inline frontend/src/component/ListItem.js:94]:
```suggestion

```

File: frontend/src/component/SpotList.js [hunk 14]
```diff
@@ -14,9 +15,11 @@ export default function SpotList({filterCountry, surfSpots}){
         <Wrapper>
             <ul>
                 {surfSpots.map((spot) => (
-                    <li>
-                        <SpotItem spot ={spot}/>
-                    </li>
+                    <a href={"/"+spot.id} >
```
Comment (jamarob) [inline frontend/src/component/SpotList.js:18]: Semantically only `li` are allowed as direct children of `ul`. You are missing a `key` here. If this is a link inside your app use react routers `Link`!

File: frontend/src/hooks/useSrufSpots.js [hunk 15]
```diff
@@ -0,0 +1,15 @@
...
+    const [surfSpots, setSurfSpots] = useState([])
+
+     useEffect(() => {
+        axios.get("/api/surfspots/all")
+            .then((response) => response.data)
+            .then(data => setSurfSpots(data))
+            .catch(error => console.error(error))
+    }, []);
+
+    return {surfSpots}
```
Comment (jamarob) [inline frontend/src/hooks/useSrufSpots.js]: When you return only one value, there is no need to wrap it in an object

File: frontend/src/pages/SearchMapPage.js [hunk 16]
```diff
@@ -1,20 +1,31 @@
...
     zoomControl: true,
-
 };
+const center = {
+    lat:50.9632238,
+    lng:6.9369613
+}
+
+export default function SearchMapPage({surfSpots}) {
+    // const {surfSpots} = useSurfSpot();
```
Comment (jamarob) [inline frontend/src/pages/SearchMapPage.js:20]:
```suggestion

```

File: frontend/src/pages/SearchMapPage.js [hunk 17]
```diff
@@ -32,14 +43,13 @@ export default function SearchMapPage() {
     if(loadError) return "Error loading Maps";
     if(!isLoaded) return "Loading Maps";
     console.log(React.version)
+    console.log(selected)
```
Comment (jamarob) [inline frontend/src/pages/SearchMapPage.js:46]:
```suggestion

```

File: frontend/src/pages/SearchMapPage.js [hunk 18]
```diff
@@ -32,14 +43,13 @@ export default function SearchMapPage() {
...
+
     const mapContainerStyle = {
         width:"100vw",
         height: "100vh"
     };
```
Comment (jamarob) [inline frontend/src/pages/SearchMapPage.js:51]: Can be moved outside the function, there is no need to redefine it on every update

File: frontend/src/pages/SearchPage.js [hunk 19]
```diff
@@ -1,17 +1,18 @@
...
+export default function SearchPage({surfSpots}){
 
         const classes = useStyles();
-        const [filterCountry, setFilterCountry] = React.useState({
+        const [filterCountry, setFilterCountry] = useState({
             name: '',
         });
-
-        const [surfSpots, setSurfSpots] = useState([])
+        // const {surfSpots} = useSurfSpot();
```
Comment (jamarob) [inline frontend/src/pages/SearchPage.js]:
```suggestion

```

File: frontend/src/pages/SearchPage.js [hunk 20]
```diff
@@ -22,28 +23,28 @@ export default function SearchPage(){
...
+    // const longitude = surfSpots[0]?.spotLocationDetails.longitude;
+    // const latitude = surfSpots[0]?.spotLocationDetails.latitude;
+    //
+    // const buttonClick =(event) => {
+    //     axios.get("/api/stormglass",
+    //     {params: {longitude, latitude}})
+    //         .then(response => response.data)
+    //         .then(data => console.log(data))
+    //         .catch(error => console.error(error))
+    // }
```
Comment (jamarob) [inline frontend/src/pages/SearchPage.js]:
```suggestion

```

File: frontend/src/pages/SearchPage.js [hunk 21]
```diff
@@ -22,28 +23,28 @@ export default function SearchPage(){
...
-            .then(data => console.log(data))
-            .catch(error => console.error(error))
+        history.push("/map");
     }
 
-
     return(
-        <Wrapper>
+        <Wrapper >
+            <div>
```
Comment (jamarob) [inline frontend/src/pages/SearchPage.js:35]: Is the div necessary?

File: frontend/src/pages/SearchPage.js [hunk 22]
```diff
@@ -22,28 +23,28 @@ export default function SearchPage(){
...
+                <Button component={Link} to={"/map"}>go to maps</Button>
+            </div>
             <FormControl className={classes.formControl}>
                 <InputLabel htmlFor="age-native-simple">Country</InputLabel>
```
Comment (jamarob) [inline frontend/src/pages/SearchPage.js:39]: Where is the element with id `age-native-simple`?

File: frontend/src/pages/SearchPage.js [hunk 23]
```diff
@@ -72,7 +71,11 @@ export default function SearchPage(){
 }
 
 const Wrapper = styled.div`
-  
+  button{
+    container:flex;
+    flex-flow: row-reverse;
+    
+  }
```
Comment (jamarob) [inline frontend/src/pages/SearchPage.js:68]: Not sure, but I think this can be deleted

**Review by jamarob — COMMENTED** [review]
Nice 🏄 Left some comments.

Please delete the comments and commented code fragments and while you are at it, remove the console.logs as well ;)

File: backend/src/main/java/de/neuefische/backend/service/SGApiService.java (hunk 1 above)
Comment (mchristia) [reply backend/src/main/java/de/neuefische/backend/service/SGApiService.java]: done :)

File: frontend/src/component/DeatailsTable.js (hunk 5 above)
Comment (mchristia) [reply frontend/src/component/DeatailsTable.js:33]: want to leave this and see if I can improve it later

**Pull request merged**